### PR TITLE
[DC-703] Add cleaning rule from DC-537 to the pipeline

### PR DESCRIPTION
**Pull request by dcampbell-vumc**
Created `null_concept_ids_for_nmeric_ppi.py` module. Created `null_concept_ids_for_nmeric_ppi_test.py` unit and integration tests.

### CODE REVIEW

File: data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+SELECT *
+FROM 
+    `{project}.{dataset}.observation`
+WHERE
+    questionnaire_response_id IS NOT NULL
+AND 
+    value_as_number IS NOT NULL
+AND 
+    (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL))
+"""
```
Comment (amcgrenera-vumc) [inline data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py:36]: Please update queries using Jinja, per the discussion in the eng sync. Here is an example Lauren sent me:

https://github.com/all-of-us/curation/blob/develop/tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/bigquery_tests_base.py#L44

Comment (dcampbell-vumc) [reply data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py:36]: Should I go ahead and set up my Jinja environment in my module? I know we discussed putting it in a utils file.

File: data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    NULL AS value_as_string)
+FROM `{project}.{dataset}.observation` AS obs 
+WHERE 
+   questionnaire_response_id IS NOT NULL
+AND
+   value_as_number IS NOT NULL
+AND
+   (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL)
+"""
+
```
Comment (amcgrenera-vumc) [inline data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py:52]: When running this query I realized this is only selecting the 1m+ rows that are effected by the where clause. I asked in the eng channel about select all or select each column individually and Krishna suggested selecting each column individually, so I think the query below should work better:

SELECT
  observation_id,
  person_id,
  observation_concept_id,
  observation_date,
  observation_datetime,
  observation_type_concept_id,
  value_as_number,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_as_string,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_as_concept_id,
  qualifier_concept_id,
  unit_concept_id,
  provider_id,
  visit_occurrence_id,
  observation_source_value,
  observation_source_concept_id,
  unit_source_value,
  qualifier_source_value,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_source_concept_id,
  CASE
    WHEN value_as_number IS NOT NULL AND value_source_concept_id IS NOT NULL AND (value_source_concept_id IS NOT NULL OR value_as_concept_id IS NOT NULL) THEN NULL
END
  AS value_source_value,
  questionnaire_response_id
FROM
  `{project}.{dataset}.observation`

File: data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py (hunk 1 above)
Comment (amcgrenera-vumc) [reply data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py:36]: Just checked with Lauren, she suggests the following, so I would go ahead and set up the Jinja env in your own module for now: 

> good question.  if you want to set up the environment for yourself in your own module, then you could go ahead and write your queries with a template the way you expect.  when we implement the utility, you could then just import the environment and remove the environment you created from your module without needing to change your sql and go through another round of SQL reviews.  so I would probably do that for now.

**Comment by amcgrenera-vumc** [comment]
Hey Danyel, everything looks good to me with this PR once you update the queries using Jinja and update `clean_numeric_ppi_query`. I compared this cleaning rule to other cleaning rules that are utilizing the base cleaning rule as I was unfamiliar, this helped me better understand the base cleaning rule, so thank you!

File: data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py (hunk 1 above)
Comment (dcampbell-vumc) [reply data_steward/cdr_cleaner/cleaning_rules/null_concept_ids_for_numeric_ppi.py:36]: Okay, I will set it up globally for now.

**Pull request closed without merging**